clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -432,6 +432,24 @@ public ConfigDef withClientSaslSupport() {
     }
 
     /**
+     * Filter configs that are not defined in this configuration definition.
+     *
+     * @param props The configs to be filtered.
+     * @return Filtered configs.
+     */
+    public Map<String, Object> filter(Map<String, Object> props) {
```

Comment:
`filter` takes a predicate where here we are passing a value. I think I would expose the known config keys from `ConsumerConfig` and `ProducerConfig` instead. And then `StreamsConfig` can do whatever it wants with it.